packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx

```diff
@@ -0,0 +1,405 @@
+          : client.login(email, password)
+      );
+      window.location.reload();
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+      setError(e.description || e.message || String(e) || "An error occurred");
+      setLoading(false);
+    }
+  }, [client, email, password, isSignup]);
+
+  const onClickGoogle = useCallback(async () => {
+    setMessage(null);
+    setError(null);
+    client.socialLogin("google-oauth2");
+  }, [client]);
+
+  const onClickFacebook = useCallback(async () => {
+    setMessage(null);
+    setError(null);
+    client.socialLogin("facebook");
+  }, [client]);
+
+  const onForgotPassword = useCallback(async () => {
+    setError(null);
+    setMessage(null);
+    try {
+      setResetPasswordLoading(true);
+      await client.resetPassword(email);
+      setMessage("Password reset email sent");
+    } catch(e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+      setError(e.description || e.message || String(e) || "An error occurred");
+    }
+    setResetPasswordLoading(false);
+  }, [client, email]);
+
+  const onLinkToLogin = useCallback(() => {
+    setAction("login");
+  }, [setAction]);
+
+  const onLinkToSignup = useCallback(() => {
+    setAction("signup");
+  }, [setAction]);
+
+  const onClose = useCallback(() => {
+    setAction(null);
+  }, [setAction]);
+
+  useEffect(() => {
+    if (!open) {
+      setEmail("");
+      setPassword("");
+      setShowPassword(false);
+      setLoading(false);
+      setMessage(null);
+      setError(null);
+    }
+  }, [open]);
+
+  const title = isSignup
+    ? "Sign up to get more from the EA Forum"
+    : "Welcome back";
+
+  const canSubmit = !!email && !!password && !loading;
+
+  const {BlurredBackgroundModal, ForumIcon, EAButton, Loading} = Components;
+  return (
+    <BlurredBackgroundModal open={open} onClose={onClose} className={classes.root}>
+      <ForumIcon
+        icon="Close"
+        onClick={onClose}
+        className={classes.close}
+      />
+      <div className={classes.lightbulb}>{lightbulbIcon}</div>
+      <div className={classes.title}>{title}</div>
+      <div className={classes.formContainer}>
+        <form onSubmit={onSubmit} className={classes.form}>
+          <input
+            type="text"
+            placeholder="Email"
+            value={email}
+            onChange={onChangeEmail}
+            className={classes.input}
+            autoFocus
+          />
+          <div className={classes.showPasswordContainer}>
+            <input
+              type={showPassword ? "text" : "password"}
+              placeholder="Password"
+              value={password}
+              onChange={onChangePassword}
+              className={classes.input}
+            />
+            <ForumIcon
+              icon={showPassword ? "EyeSlash" : "Eye"}
+              onClick={toggleShowPassword}
+              className={classes.showPasswordButton}
+            />
```

Comment:
nit (for @agnesstenlund ): the 1password icon covers the show/hide icon
<img width="380" alt="Screenshot 2024-01-30 at 17 45 13" src="https://github.com/ForumMagnum/ForumMagnum/assets/77623106/8607ca05-bba5-427f-8a95-e3aed04be700">

Reply:
In the non-integrated version this does work, probably the icon just needs to be moved out of the `<input>`

<img width="472" alt="Screenshot 2024-01-30 at 18 43 02" src="https://github.com/ForumMagnum/ForumMagnum/assets/77623106/6fa2b5d1-8d77-4a7e-bf0b-78736f171bc1">